### DISCUSSION TITLE
기존에 존재하는 구간 사이에 새로운 구간을 추가하는 작업

### DISCUSSION THREAD

**Discussion by lxxjn0** [dev-floor/javascript-subway-path#1] [General]
### 기존에 존재하는 구간 사이에 새로운 구간을 추가할 경우

`교대 - 2km / 3분 - 역삼` 구간 사이에 새로운 구간을 추가할 경우 **상행역**을 기준으로만 추가 가능하도록 하겠습니다!

즉, `교대 - 2km / 3분 - 강남` 구간으로는 **추가 가능**하고 `강남 - 2km / 3분 - 역삼` 구간으로는 **추가가 불가능**하도록 설정하도록 하겠습니다.

~~물론, 두 가지 상황 전부 다 가능하게 해도 괜찮은데 그러면 너무 코드가 복잡해질 것 같아서 우선은 위의 제약 조건으로 진행하되, 본인이 구현할 수 있다면 모든 상황에 대해 추가 가능하도록 구현해도 상관 없습니다~~ 💯 

아래 코멘트를 남겨뒀으니 참고 부탁드려요 👍

**Comment by lxxjn0** [comment]
즉, 구간이 `A - B`가 있으면 `B`가 새롭게 추가될 역이라고 생각하면 될 것 같습니다. `A`는 이전역을, `B`는 새롭게 추가할 역을 말합니다.

`A - C` 구간을 `A - B - C`라는 구간으로 만들고자 한다면 새롭게 추가되는 역이 `B` 역이고, 이전역이 `A` 역이므로 `A - B` 구간을 추가하면 됩니다.

이는 삭제하는 경우에도 동일하게 적용하도록 하겠습니다!!

**Reply by Derek-94** [reply]
아아 그러니까 `A - C` 구간에서, 중간에 `B` 역을 추가하고 싶다면, `B - C` 로는 추가가 불가능하고, `A - B` 로 추가만이 허용된다는 뜻인가요?! 🤔

**Reply by lxxjn0** [reply]
넵!! 💯

**Comment by lxxjn0** [comment]
위와 같은 방법을 사용할 경우 맨 앞 역을 추가할 수 없게 됩니다.

왜냐하면 만약 `A - B - C` 구간에 `D`역을 `A` 앞에 두고 싶다면 `(공백) - D`와 같은 방법으로 (이전역이 존재하지 않으므로) 추가해야 하는데 현재 요구 조건에서 역의 이름의 길이는 2자 이상만 가능하기 때문입니다.

그래서 일단은 '가장 앞 역(상행 종점)의 경우 추가할 수 없다'라고 가정하도록 하겠습니다.

만약, 맨 앞 역(상행 종점)을 추가하는 로직을 구현할 경우 Empty Station을 만들어 두고 이를 사용하여 `(empty station) - A - B` 와 같은 식으로 구현하면 될 것 같습니다.
